controllers/replica_set_controller.go

```diff
@@ -457,14 +471,15 @@ func buildService(mdb mdbv1.MongoDBCommunity) corev1.Service {
 		Build()
 }
 
-// validateUpdate validates that the new Spec, corresponding to the existing one
-// is still valid. If there is no a previous Spec, then the function assumes this is
-// the first version of the MongoDB resource and skips.
-func (r ReplicaSetReconciler) validateUpdate(mdb mdbv1.MongoDBCommunity) error {
+// validateSpec checks if MongoDB resource Spec is valid.
+// If there is no a previous Spec, then the function assumes this is the first version
+// and runs the initial spec validations otherwise it validates that the new Spec,
+// corresponding to the existing one is still valid
+func (r ReplicaSetReconciler) validateSpec(mdb mdbv1.MongoDBCommunity) error {
 	lastSuccessfulConfigurationSaved, ok := mdb.Annotations[lastSuccessfulConfiguration]
 	if !ok {
-		// First version of Spec, no need to validate
-		return nil
+		// First version of Spec
+		return validation.ValidateInitalSpec(mdb)
```

Comment:
This validation will only apply a single time, (the first reconciliation) so we'll be skipping this reconciliation every other time. The other validation (ValidateUpdate) only makes sense to happen all times but the first, but here I believe we want to check this every reconciliation.

Reply:
you're right since users can be also added after the initial reconcile process was successful.
I made the small change to validate users in both scenarios